src/realm.js

```diff
@@ -1528,19 +1528,12 @@ export class Realm {
   recordModifiedBinding(binding: Binding, value?: Value): Binding {
     const isDefinedInsidePureFn = root => {
       let context = this.getRunningContext();
-      let { lexicalEnvironment: env, function: func } = context;
-
-      invariant(func instanceof FunctionValue);
-      if (root instanceof FunctionEnvironmentRecord && func === root.$FunctionObject) {
-        return true;
-      }
-      if (this.createdObjectsTrackedForLeaks !== undefined && !this.createdObjectsTrackedForLeaks.has(func)) {
-        return false;
-      }
-      env = env.parent;
+      let { lexicalEnvironment: env } = context;
       while (env) {
```

Comment:
Not your fault, but while you are here, would it be possible to change this to "env !== null". We avoid implicit conversions to boolean whenever possible.